### PR TITLE
[BI-25] Integrate BE api for register page

### PR DESCRIPTION
**Pull request by LAlex14**
image

am integrat BE pentru Register Page, in momentul in care se apasa butonul se face un request de tip post si in print screenul de mai sus s-a afisat raspunsul, in ultimul push, am sters functia de afisare a raspunsului pentru a nu fi disponibile datele utilizatorilor ce se inregistreaza

### CODE REVIEW

File: src/views/Register.vue [hunk 1]
```diff
@@ -74,6 +74,8 @@
 </template>
 
 <script>
+import authService from '../api/authService';
```
Comment (Ceciliaa) [inline src/views/Register.vue:77]: you can directly import the register method:

`import {register}  from '@/api/authService';`

File: src/views/Register.vue [hunk 2]
```diff
@@ -88,8 +90,12 @@ export default {
     };
   },
   methods: {
-    handleSubmit() {
-      console.log(this.model);
+    async handleSubmit() {
+      try {
```
Comment (Ceciliaa) [inline src/views/Register.vue:94]: Add a loading state for this page. 
When user press the register button, we should display a loading state until the response from server is received

Comment (PascuMihai) [reply src/views/Register.vue:94]:
```
async onSubmit() {
      try {
        this.loading = true
        await this.$store.dispatch('auth/login', this.model)
        await this.$router.push('/reports')
      } finally {
        this.loading = false
      }
    },
```

Comment (PascuMihai) [reply src/views/Register.vue:94]: something like that, and you can use the loading variable to disable the button in the form so you can't press it multiple times and call the BE multiple times

**Review by PascuMihai — APPROVED** [review]
nice 👍

File: src/views/Register.vue [hunk 3]
```diff
@@ -59,13 +59,17 @@
...
+            <base-button
               type="submit"
-              class="flex justify-center w-full px-4 py-2 my-4 text-sm font-medium text-white bg-indigo-600 border border-transparent rounded-md shadow-sm hover:bg-indigo-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-indigo-500"
-              :disabled="!valid"
+              :disabled="loading === valid"
+              class="flex items-center justify-center w-full px-4 py-2 my-4 text-sm font-medium text-white border border-transparent rounded-md shadow-sm"
             >
-              {{ $t('register.button.label.signUp') }}
-            </button>
+              <i
```
Comment (Ceciliaa) [inline src/views/Register.vue:67]: Put this in the BaseButton Component.
This state should be available on each button when the property loading is true.

**Review by Ceciliaa — APPROVED** [review]

**Pull request merged**